pkg/kubelet/util/util_unix.go

```diff
@@ -50,7 +51,29 @@ func CreateListener(endpoint string) (net.Listener, error) {
 		return nil, fmt.Errorf("failed to unlink socket file %q: %v", addr, err)
 	}
 
-	return net.Listen(protocol, addr)
+	// Use exponential backoff until pod-resource directory is created by kubelet.Run()
```

Comment:
Nit: In your comment, could you include the maximum amount of time we could end up waiting before failing?

Reply:
comment is updated with number tries and max wait time.